frontend/src/metabase/query_builder/components/notebook/NotebookStep/NotebookStep.tsx

```diff
@@ -0,0 +1,168 @@
+
+import type Question from "metabase-lib/Question";
+import type StructuredQuery from "metabase-lib/queries/StructuredQuery";
+
+import {
+  NotebookStep as INotebookStep,
+  NotebookStepAction,
+} from "../lib/steps.types";
+import NotebookStepPreview from "../NotebookStepPreview";
+
+import { STEP_UI } from "./steps";
+import ActionButton from "./ActionButton";
+import {
+  StepActionsContainer,
+  StepBody,
+  StepContent,
+  StepHeader,
+  StepButtonContainer,
+  StepRoot,
+} from "./NotebookStep.styled";
+
+function hasLargeButton(action: NotebookStepAction) {
+  return !STEP_UI[action.type].compact;
+}
+
+function getTestId(step: INotebookStep) {
+  const { type, stageIndex, itemIndex } = step;
+  return `step-${type}-${stageIndex || 0}-${itemIndex || 0}`;
+}
+
+interface NotebookStepProps {
+  step: INotebookStep;
+  sourceQuestion?: Question;
+  isLastStep: boolean;
+  isLastOpened: boolean;
+  reportTimezone?: string;
+  openStep: (id: string) => void;
+  updateQuery: (query: StructuredQuery) => Promise<void>;
+}
+
+function NotebookStep({
+  step,
+  sourceQuestion,
+  isLastStep,
+  isLastOpened,
+  reportTimezone,
+  openStep,
+  updateQuery,
+}: NotebookStepProps) {
+  const [isPreviewOpen, { turnOn: openPreview, turnOff: closePreview }] =
+    useToggle(false);
+
+  const actionButtons = useMemo(() => {
+    const actions = [];
+    const hasLargeActionButtons =
+      isLastStep && step.actions.some(hasLargeButton);
+
+    actions.push(
+      ...step.actions.map(action => {
+        const stepUi = STEP_UI[action.type];
+        return {
+          priority: stepUi.priority,
+          button: (
+            <ActionButton
+              key={`actionButton_${stepUi.title}`}
+              mr={isLastStep ? 2 : 1}
+              mt={isLastStep ? 2 : undefined}
+              color={stepUi.getColor()}
+              large={hasLargeActionButtons}
+              {...stepUi}
+              onClick={() => action.action({ query: step.query, openStep })}
+            />
+          ),
+        };
+      }),
+    );
+
+    actions.sort((a, b) => (b.priority || 0) - (a.priority || 0));
+
+    return actions.map(action => action.button);
+  }, [step.query, step.actions, isLastStep, openStep]);
+
+  const handleClickRevert = useCallback(() => {
+    const reverted = step.revert?.(step.query);
+    if (reverted) {
+      updateQuery(reverted);
+    }
+  }, [step, updateQuery]);
+
+  const {
+    title,
+    getColor,
+    component: NotebookStepComponent,
+  } = STEP_UI[step.type] || {};
+
+  const color = getColor();
+  const canPreview = step?.previewQuery?.isValid?.();
+  const hasPreviewButton = !isPreviewOpen && canPreview;
+
+  return (
+    <ExpandingContent isInitiallyOpen={!isLastOpened} isOpen>
+      <StepRoot
+        className="hover-parent hover--visibility"
+        data-testid={getTestId(step)}
+      >
+        <StepHeader color={color}>
+          {title}
+          <Icon
```

Comment:
I recently opened an issue (https://github.com/metabase/metabase/issues/29276) describing how this button doesn't work for the data section (which makes sense since there's no reason to completely remove data).

Would you be interested in fixing that in this PR? A simple fix would be to not render it for the data section.

```typescript
{step.type !== "data" &&
<Icon...
```

Reply:
@EmmadUsmani, #29575 should fix this